src/neuroconv/datainterfaces/ecephys/edf/edfdatainterface.py

```diff
@@ -0,0 +1,61 @@
+"""Authors: Heberto Mayorquin"""
+from spikeinterface.extractors import EDFRecordingExtractor
+from ..baserecordingextractorinterface import BaseRecordingExtractorInterface
+
+from ....utils.types import FilePathType
+
+try:
+    import pyedflib
+
+    HAVE_PYEDFLIB = True
+except ImportError:
+    HAVE_PYEDFLIB = False
+INSTALL_MESSAGE = "Please install pyedflib (https://pypi.org/project/pyEDFlib/) to use this interface!"
+
+
+class EDFRecordingInterface(BaseRecordingExtractorInterface):
+    """Primary data interface class for converting European Data Format (EDF) data."""
+
+    RX = EDFRecordingExtractor
+
+    def __init__(self, file_path: FilePathType, verbose: bool = True):
+        """
+        Load and prepare data for EDF
+        Currently only continuous EDF+ files (EDF+C) and original EDF files (EDF) are supported
+
+
+        Parameters
+        ----------
+        folder_path: str or Path
+            Path to the edf file
+        verbose: bool, True by default
+            Allows verbose.
+        """
+        assert HAVE_PYEDFLIB, INSTALL_MESSAGE
+        super().__init__(file_path=file_path, verbose=verbose)
+        self.edf_header = self.recording_extractor.neo_reader.edf_header
+
+    def extract_nwb_file_metadata(self):
+
+        nwbfile_metadata = dict(
+            session_start_time=self.edf_header["startdate"], experimenter=self.edf_header["technician"]
+        )
+
+        return nwbfile_metadata
+
+    def extract_subject_metadata(self):
+
+        subject_metadata = dict(subject_id=self.edf_header["patientcode"], date_of_birth=self.edf_header["birthdate"])
```

Comment:
I was wondering about this. I just changed the metadata schema for experimenters so it could accept empty strings. Maybe we want

Reply:
> Maybe we want some general code after run conversion that transforms empty strings to None so we don't have this problem for any possible field that can could be empty.

I'd rather just set them properly to begin with. That sounds like magic that could cause complications down the road.

Reply:
Fair enough. I will revert that.

Reply:
I don't understand why you would make changes to the schema to allow for empty strings instead of simply changing the values here to be the proper `None` values

Reply:
It was an honest mistake. I saw that all the other string properties in the schema accepted empty strings and that experimenter is the exception because it accepts a list as well. So I went for unifying at that level. I will revert later today.

Reply:
@bendichter I reverted the changes and pushed your commits.